File: l3/z1/Okrag.java
public class Okrag extends Figura{

    Okrag(float r) {
        this.mR = r;
        this.mFiugreName = "okrąg";
    }

    @Override
    public float area() {
        return (float)(2 * Math.PI * (mR * mR));
    }

    @Override
    public float length() {
        return (float)(2 * Math.PI * mR);
    }

    public String getFigureName() {
        return mFigureName;
    }
    static boolean validate(float r) {
        return r > 0;
    }

    private float mR = 0.f;
    private String mFigureName = "";
}

File: l3/z2/Okrag.cpp
#include <cmath>
#include "Okrag.hpp"

namespace Zadanie {
    Okrag::Okrag(float r, std::string figureName)
            : mR(r), Figura(std::move(figureName)) {
    }

    float Okrag::area() {
        return static_cast<float>(M_PI * mR * mR);
    }

    float Okrag::length() {
        return static_cast<float>(M_PI * mR * 2);
    }

    bool Okrag::validate(float r) {
        if (r > 0)
            return true;
        else
            return false;
    }

}
